torch/nn/modules/utils.py

```diff
@@ -57,22 +57,23 @@ def consume_prefix_in_state_dict_if_present(
         state_dict (OrderedDict): a state-dict to be loaded to the model.
         prefix (str): prefix.
     """
-    keys = sorted(state_dict.keys())
+    keys = list(state_dict.keys())
     for key in keys:
         if key.startswith(prefix):
             newkey = key[len(prefix) :]
             state_dict[newkey] = state_dict.pop(key)
 
     # also strip the prefix in metadata if any.
-    if "_metadata" in state_dict:
-        metadata = state_dict["_metadata"]
-        for key in list(metadata.keys()):
+    if hasattr(state_dict, "_metadata"):
+        keys = list(state_dict._metadata.keys())
+        for key in keys:
             # for the metadata dict, the key can be:
             # '': for the DDP module, which we want to remove.
             # 'module': for the actual model.
             # 'module.xx.xx': for the rest.
-
             if len(key) == 0:
                 continue
-            newkey = key[len(prefix) :]
-            metadata[newkey] = metadata.pop(key)
+            # handling both, 'module' case and  'module.' cases
+            if key == prefix.replace('.', '') or key.startswith(prefix):
+                newkey = key[len(prefix) :]
+                state_dict._metadata[newkey] = state_dict._metadata.pop(key)
```

Comment:
Ah, so the 

value of `state_dict._metadata['module']` will replace the `state_dict._metadata['']` is that right?

Reply:
yes!